pkg/sql/logictest/testdata/logic_test/mixed_version_sequence_per_node_cache

```diff
@@ -0,0 +1,104 @@
+# LogicTest: cockroach-go-testserver-23.1
+
+# Verify that all nodes are running the previous version.
+
+query T nodeidx=0
+SELECT crdb_internal.node_executable_version()
+----
+23.1
+
+query T nodeidx=1
+SELECT crdb_internal.node_executable_version()
+----
+23.1
+
+query T nodeidx=2
+SELECT crdb_internal.node_executable_version()
+----
+23.1
+
+statement error pq: at or near "per": syntax error
+CREATE SEQUENCE cache_test PER NODE CACHE 10
+
+# Upgrade one node to 24.1
+
+upgrade 0
+
+# Verify that node index 0 is now running 24.1 binary.
+
+query T nodeidx=0
+SELECT crdb_internal.release_series(crdb_internal.node_executable_version())
+----
+24.1
+
+statement error pgcode 0A000 node-level cache not supported before V24.1
+CREATE SEQUENCE cache_test PER NODE CACHE 10
+
+upgrade 1
+
+upgrade 2
+
+query T nodeidx=1
+SELECT crdb_internal.release_series(crdb_internal.node_executable_version())
+----
+24.1
+
+query T nodeidx=2
+SELECT crdb_internal.release_series(crdb_internal.node_executable_version())
+----
+24.1
+
+query B retry
+SELECT crdb_internal.is_at_least_version('23.2-4')
+----
+true
+
+statement ok nodeidx=0
+CREATE SEQUENCE cache_test PER NODE CACHE 10
+
+# Verify that the sequence gets incremented to the default cache
+# size of 256 for node-level sequence caching
+subtest serial_cached_node_sequence
+
+statement ok nodeidx=0
+SET serial_normalization = sql_sequence_cached_node
+
+statement ok nodeidx=0
+CREATE TABLE serial (
+  cached SERIAL
+);
+
+statement ok nodeidx=0
+INSERT INTO serial (cached) VALUES (DEFAULT);
+
+query I nodeidx=0
+SELECT cached from serial;
+----
+1
+
+query T nodeidx=0
+SELECT pg_get_serial_sequence('serial', 'cached')
+----
+public.serial_cached_seq
+
+query I nodeidx=0
+SELECT last_value from public.serial_cached_seq;
+----
+256
+
+statement ok nodeidx=0
```

Comment:
nit: can we made this a node different than the one we inserted into above? if not, then i don't think the stuff that follows (minus DROP TABLE) is needed